File: baseproject-android/app/src/main/java/com/chang/template/activity/home/MainActivity.java
package com.chang.template.activity.home;

import android.os.Bundle;

import com.chang.template.R;
import com.chang.template.activity.BaseActivity;

import javax.inject.Inject;

public class MainActivity extends BaseActivity {

    @Inject
    MainPresenter mPresenter;

    MainMVPFragment fragment;

    @Override
    protected int getContentView() {
        return R.layout.activity_main;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        initViews();
    }

    private void initViews() {

        fragment = (MainMVPFragment) getSupportFragmentManager()
                .findFragmentById(R.id.content_view);

        if (fragment == null) {
            fragment = MainMVPFragment.newInstance();

            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.content_view, fragment)
                    .commit();
        }

        DaggerMainComponent.builder()
                .mainPresenterModule(new MainPresenterModule(this, fragment))
                .build()
                .inject(this);

    }

}

File: baseproject-android/app/src/main/java/com/chang/template/activity/home/MainActivity.kt
package com.chang.template.activity.home

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import com.chang.template.R
import com.chang.template.activity.BaseActivity
import com.chang.template.utils.replaceFragmentInActivity
import java.io.Serializable

class MainActivity : BaseActivity() {

    private lateinit var mainFragment: MainFragment
    private lateinit var data: MainData

    override val contentView: Int
        get() = R.layout.activity_main

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        getData()
        initViews()
    }

    private fun getData() {
        data = intent.getSerializableExtra(DATA) as MainData
    }

    private fun initViews() {

        mainFragment = supportFragmentManager.findFragmentById(R.id.content_view) as MainFragment?
                ?: MainFragment.newInstance().also {
                    replaceFragmentInActivity(it, R.id.content_view)
                }

        MainPresenter(this, mainFragment, data)

    }

    companion object {

        fun launch(activity: Activity) {
            Intent(activity, MainActivity::class.java)
                    .apply { putExtra(DATA, MainData() as Serializable) }
                    .apply { activity.startActivity(this) }
        }

    }

}
